notebooks/transport/http.go

```diff
@@ -0,0 +1,196 @@
+package transport
+
+import (
+	"net/http"
+	"strconv"
+
+	"github.com/influxdata/influxdb/v2"
+	feature "github.com/influxdata/influxdb/v2/kit/feature"
+	"github.com/influxdata/influxdb/v2/kit/platform"
+	"github.com/influxdata/influxdb/v2/kit/platform/errors"
+	kithttp "github.com/influxdata/influxdb/v2/kit/transport/http"
+
+	"github.com/go-chi/chi"
+	"github.com/go-chi/chi/middleware"
+	"go.uber.org/zap"
+)
+
+// NotebookHandler is the handler for the notebook service
+type NotebookHandler struct {
+	chi.Router
+
+	api *kithttp.API
+	log *zap.Logger
+}
+
+const (
+	prefixNotebooks = "/api/v2private/flows"
+)
+
+func NewNotebookHandler(log *zap.Logger) *NotebookHandler {
+	h := &NotebookHandler{
+		log: log,
+		api: kithttp.NewAPI(kithttp.WithLog(log)),
+	}
+
+	r := chi.NewRouter()
+	r.Use(
+		middleware.Recoverer,
+		middleware.RequestID,
+		middleware.RealIP,
+		h.notebookFlag, // temporary, remove when feature flag for notebooks is removed
+	)
+
+	r.Route("/orgs/{orgID}/flows", func(r chi.Router) {
+		r.Get("/", h.handleGetNotebooks)
+		r.Post("/", h.handleCreateNotebook)
+
+		r.Route("/{id}", func(r chi.Router) {
+			r.Get("/", h.handleGetNotebook)
+			r.Patch("/", h.handlePatchNotebook)
+			r.Delete("/", h.handleDeleteNotebook)
+		})
+	})
+
+	h.Router = r
+
+	return h
+}
+
+func (h *NotebookHandler) Prefix() string {
+	return prefixNotebooks
+}
+
+// notebookFlag is middleware for returning no content if the notebooks feature
+// flag is set to false. remove this middleware when the feature flag is removed.
+func (h *NotebookHandler) notebookFlag(next http.Handler) http.Handler {
+	fn := func(w http.ResponseWriter, r *http.Request) {
+		flags := feature.FlagsFromContext(r.Context())
+
+		if !flags["notebooks"].(bool) {
+			h.api.Respond(w, r, http.StatusNoContent, nil)
+			return
+		}
+
+		next.ServeHTTP(w, r)
+	}
+
+	return http.HandlerFunc(fn)
+}
+
+// get a list of all notebooks for an org
+func (h *NotebookHandler) handleGetNotebooks(w http.ResponseWriter, r *http.Request) {
+	orgID, err := getIDfromReq(r, "orgID")
+	if err != nil {
+		h.api.Err(w, r, err)
```

Comment:
Should the error cases here be using `errors.Error`?

Reply:
`err` comes from `errors.Error` eventually, in the `getIDfromReq` function. I noticed that the error messages could be a little more helpful so I made some improvements there - the response error message will differentiate between a problem with the `orgID` parameter and the `id` parameter.